docs/user/device-api-statuses.md

```diff
@@ -0,0 +1,126 @@
+
+## Device Status
+
+The Device Status reflects the availability and health of the device's hardware and operating system.
+
+The `device.status.summary` field can have the following values:
+
+| Status | Description | Formal Definition |
+| ------ | ----------- | ----------------- |
+
+## Device Update Status
+
+The Device Update Status reflects whether the device's currently running specification (OS, configuration, applications, etc.) matches the user's intent as expressed via the device spec or the fleet's device template.
+
+The `device.status.updated.status` field can have the following values:
+
+| Status | Description | Formal Definition |
+| ------ | ----------- | ----------------- |
+| `UpToDate` | The device is running according to its device spec. If the device is part of a fleet, that device spec is at the same version as the fleet's device template. | `deviceMeetsDeviceSpec && (deviceIsNotFleetManaged \|\| deviceSpecAtFleetSpec)` |
+| `Updating` | The device is in the process of updating to its device spec. | `deviceIsUpdating` |
+| `OutOfDate` | The device is not `UpToDate` and not `Updating`. | |
+| `Unkown` | The device's agent either never reported status or its last reported status was `Updating` and the device has been disconnected since. | |
+
+whereby the formal definition uses the following helper definitions:
+
+```golang
+// a device is assumed disconnected if its agent hasn't sent an update for the duration of a disconnectionTimeout
+deviceIsDisconnected := device.status.lastSeen + disconnectionTimeout < time.Now()
+
+// a device is not managed by a fleet if its owner field is unset
+deviceIsNotFleetManaged := len(device.metadata.owner) == 0
+
+// a device is updating when the agent sets the "Updating" condition to true
+deviceIsUpdating := device.status.conditions.updating == true
+
+// a device meets its device spec when the version of the device spec that the agent reports as running
+// equals the version rendered to the device by the service
+deviceMeetsDeviceSpec := device.status.config.renderedVersion == device.metadata.annotations.renderedVersion
+
+// a device spec is up-to-date relative to the fleet's device template if both are of the same template version.
+deviceSpecAtFleetSpec := device.metadata.annotations.templateVersion == fleet[device.metadata.owner].spec.templateVersion
+```
+
+The `device.status.updated.???` field contains the current state of the update in progress and can take the following values:
+
+| Update State | Description |
+| ------------ | ----------- |
+| `Preparing` | The agent is validating the desired device spec and downloading dependencies. No changes have been made to the device's configuration yet. |
+| `ReadyToUpdate` | The agent has validated the desired spec, downloaded all dependencies, and is ready to update. No changes have been made to the device's configuration yet. |
```

Comment:
I guess the device will be in this state while waiting for its maintenance window?  Otherwise it will immediately move to ApplyingUpdate and this state isn't needed.

Reply:
Yes, that'd be the idea. I'd like to get alignment on the states and behaviour, even if currently that state doesn't exist.